### ISSUE TITLE
Sudden influx of TimoutException using V2 Remoting 

### ISSUE THREAD

**Issue by gparrington** [Azure/service-fabric-issues#1055]
Our production fabric apps (which have been running for over a year using V1 remoting) was recently upgraded to use V2. Although the upgrade process appeared to go fine two issues occurred

1. After the upgrade we saw exceptions in the app insights logs reporting 'V2 Endpoint not found' for some Stateful services - despite that fact that the upgrade appeared to have completed OK and the fabric explorer showed green and all service/applications as OK. We had to restart every node in the cluster using the fabric explorer to clear these errors.

2. After the cluster had been running fine for several days we suddenly began to receive Timeout exceptions on all calls between our services that were initiated using ServiceProxy.Create(). Our logs showed no other errors, the explorer showed everything as OK and cpu load on the affected nodes was < 10%. Once again only a restart of the affected nodes would clear this error.

Both these incidents caused downtime for our customers and a significant influx of support calls - what's the likely cause and how can we avoid them in the future

**Comment by gparrington** [comment]
Hi @suchiagicha - we performed the steps exactly as described in the docs at https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication-remoting#how-to-upgrade-from-remoting-v1-to-remoting-v2. Since we have multiple services we made sure to perform these steps on all of them and then went through a standard upgrade deployment of our application. We had to use compatability mode as deployment consists of multiple independently deployed services and takes several hours to complete meaning that during the upgrade we would still have client services using V1 for a significant period of time

We actually performed this first on a staging environment (which showed no issues) before we repeated the upgrade on our production fabric (using exactly the same packages) which is were we ran into issues immediately after upgrade and then again a few days later as described. The services were built using SDK version 6.0.232/2.8.232

**Comment by suchiagicha** [comment]
Did you make sure Client Service (Code which create ServcieProxy) needs to be upgraded after Remoting Service (where listener resides). If not. then "V2 Endpoint errors will appear".
Reason is you client is using remoting v2 but your service has not yet created v2 listener.
Above steps needs to be followed in sequence.

**Comment by gparrington** [comment]
Yes. We were very careful to upgrade the back-end services with support for V1&V2 (via listener.compat) before we rolled out the clients (which simply upgraded from V1 to V2).

Right now I think our major concern is why we started to get TimeOutExceptions after several days of normal running

**Comment by suchiagicha** [comment]
For the timeout, can check the traces, Could you share the SF traces here?

**Comment by gparrington** [comment]
Sure if you tell me how to collect them - I've used CollectGuestLogs.exe on the nodes to gather all the logs together - is that all you need?. Which do you want and where should I upload them to?

**Comment by suchiagicha** [comment]
In your clustermanifest , there will be parameter with name FabricLogRoot which will tell you the location for the logs. You can get for the few machines where you saw the timeout.  You can create zip and upload here or send me at suchia@microsoft.com

**Comment by gparrington** [comment]
Not according to our manifest (attached) which I extracted from our production cluster in Azure using the fabric explorer. If I read it right the logs are going to blob/table storage

[ClusterManifest.zip](https://github.com/Azure/service-fabric-issues/files/2016089/ClusterManifest.zip)

**Comment by suchiagicha** [comment]
PS : To investigate the issues, could you please open a support ticket from Azure portal, which will enable us to give you a faster resolution and complete postmortum on this issue. Please open a ticket using https://portal.azure.com/?#blade/Microsoft_Azure_Support/HelpAndSupportBlade  
Please refer https://azure.microsoft.com/en-us/support/options/ for details on support options  

Are you using onprem cluster or you are using Azure cluster?
If Azure cluster, then you can tell em the resource Id and region.

If onprem, give the table storage details.

**Comment by suchiagicha** [comment]
@gparrington 
Also could you share the servicemanifest for the remoting service?
